code/game/objects/items/weapons/material/coins.dm

```diff
@@ -58,29 +58,69 @@
 	if(!can_flip)
 		to_chat(user, SPAN_WARNING("\The [src] is already being flipped!"))
 		return
-	coin_flip(user)
+	coin_flip(user, 0)
 
-/obj/item/coin/proc/coin_flip(var/mob/user)
+/obj/item/coin/throw_impact(atom/hit_atom, speed, user)
+	..()
+	coin_flip(user, 1)
 
+/obj/item/coin/proc/coin_flip(var/mob/user, thrown, rigged = FALSE)
 	if(!can_flip)
 		return
 
-
 	var/decl/currency/cur = decls_repository.get_decl(currency)
 	var/datum/denomination/denomination = cur.denominations_by_value[currency_worth]
 
 	if(!denomination || !length(denomination.faces))
-		to_chat(user, SPAN_WARNING("\The [src] is not the right shape to be flipped."))
+		if(user)
+			to_chat(user, SPAN_WARNING("\The [src] is not the right shape to be flipped."))
 		return
 
 	can_flip = FALSE
-	playsound(usr.loc, 'sound/effects/coin_flip.ogg', 75, 1)
-	user.visible_message(SPAN_NOTICE("\The [user] flips \the [src] into the air."))
+
+	var/matrix/flipit = matrix()
+	flipit.Scale(0.2, 1)
+	animate(src, transform = flipit, time = 2, easing = QUAD_EASING)
+	flipit.Scale(5, 1)
+	flipit.Invert()
+	flipit.Turn(rand(1, 359))
+	animate(src, transform = flipit, time = 2, easing = QUAD_EASING)
+	flipit.Scale(0.2, 1)
+	animate(src, transform = flipit, time = 2, easing = QUAD_EASING)
+
+	flipit.Scale(5, 1)
+	if(pick(0, 1))
+		flipit.Invert()
+	flipit.Turn(rand(1, 359))
+	animate(src, transform = flipit, time = 2, easing = QUAD_EASING)
+
+	if (prob(0.1) || rigged)
```

Comment:
Doesn't this mean a coin will now always land on the edge forever?

Reply:
nope, not always. `rigged` turned off by default and needed for special actions like cheating :^)

Reply:
Oh, this is for a future feature? I didn't see `rigged` was an argument not a member variable.